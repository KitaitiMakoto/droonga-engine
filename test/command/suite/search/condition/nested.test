#@require-catalog-version 2
#@disable-logging
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.1",
    "values": {
      "title": "Groonga overview"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.2",
    "values": {
      "title": "Full text search and Instant update"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.3",
    "values": {
      "title": "Column store and aggregate query"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.4",
    "values": {
      "title": "Inverted index and tokenizer"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.5",
    "values": {
      "title": "Sharable storage and read lock-free"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.6",
    "values": {
      "title": "Geo-location (latitude and longitude) search"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.7",
    "values": {
      "title": "Groonga library"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.8",
    "values": {
      "title": "Groonga server"
    }
  }
}
{
  "dataset": "Droonga",
  "type": "add",
  "body": {
    "table": "Sections",
    "key": "1.9",
    "values": {
      "title": "Groonga storage engine"
    }
  }
}
#@enable-logging
{
  "type": "search",
  "dataset": "Droonga",
  "body": {
    "queries": {
      "result": {
        "source": "Sections",
        "condition": [
          "&&",
          "title @ 'groonga'",
          "title @ 'engine'"
        ],
        "sortBy": ["_key"],
        "output": {
          "elements": [
            "count",
            "attributes",
            "records"
          ],
          "limit": -1,
          "attributes": ["title"]
        }
      }
    }
  }
}
