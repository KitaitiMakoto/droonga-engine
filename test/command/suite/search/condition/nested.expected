{
  "inReplyTo": "request-id",
  "statusCode": 200,
  "type": "search.result",
  "body": {
    "result": {
      "count": 1,
      "records": [
        [
          "Groonga storage engine"
        ]
      ]
    }
  }
}
